iot-lambda-pub-receiver-cdk/lib/lambda-iot-cdk-stack.ts

```diff
@@ -0,0 +1,115 @@
+      }
+    })
+    iotPubLambda.addToRolePolicy(iotPubPermission)
+
+    // Receiver lambda function
+    const iotReceiverPermission = new PolicyStatement(({
+      effect: Effect.ALLOW,
+      resources: [ `arn:aws:iot:${mqttTopicRegion}:${mqttTopicAccount}:topic/${mqttTopicName}` ],
+      actions: [
+        "iot:Receive"
+      ]
+    }));
+    const iotReceiverLambda = new Function(this, 'iotReceiverHandler', {
+      handler: 'receiver_function.handler',
+      description: 'This function get invoked by AWS IoT Core through the action-rule',
+      code: Code.fromAsset('./src', {
+        bundling: {
+          image: Runtime.PYTHON_3_12.bundlingImage,
+          command: [
+            'bash', '-c',
+            'pip install -r receiver_requirements.txt -t /asset-output && cp -au . /asset-output'
+          ],
+        },
+      }),
+      runtime: Runtime.PYTHON_3_12,
+      architecture: Architecture.ARM_64,
+      logGroup: this.addLogGroup(`/aws/lambda/receiver-lambda`)
+    })
+    iotReceiverLambda.addToRolePolicy(iotReceiverPermission)
+
+    // Topic rule
+    const errorLogGroup = new LogGroup(this, 'RuleErrorLogGroup', {
+      logGroupName: '/aws/iot/rule-error-logs',
+      retention: RetentionDays.FIVE_DAYS,
+      removalPolicy: RemovalPolicy.DESTROY 
+    })
+    let topicRule = new TopicRule(this, 'IoTTopicRule', {
+      topicRuleName: 'ProcessIoTMessages',
+      description: 'Invokes the lambda function',
+      sql: IotSql.fromStringAsVer20160323("SELECT * FROM 'my/mqtt/topic'"),
+      actions: [ new LambdaFunctionAction(iotReceiverLambda) ],
+      errorAction: new CloudWatchLogsAction(errorLogGroup)
+    })
+
+    // Grant permission for AWS IoT to invoke the Lambda function
+    const iotServicePrincipal = new ServicePrincipal('iot.amazonaws.com');
+    iotReceiverLambda.grantInvoke(iotServicePrincipal);
+
+    // Outputs
+    new CfnOutput(this, "IoT Endpoint Address", {
+      value: iotEndpointAddress ?? "Error: can't get the IoT Endpoint Address!",
+    });
+  }
+
+  // Utility function to return a log-group object
+  private addLogGroup(logGroupName: string) {
+    const retentionDays = RetentionDays.FIVE_DAYS
+    const removalPolicy = RemovalPolicy.DESTROY
+    const props = { logGroupName, retentionDays, removalPolicy }
+    return new LogGroup(this, `${logGroupName}`, props)
+  }
+
+  // Get the current account IoT-Endpoint 
+  private getIoTEndpoint() {
+    const ioTEndpoint = new AwsCustomResource(this, 'IoTEndpoint', {
+      onCreate: {
+          service: 'Iot',
+          action: 'describeEndpoint',
+          physicalResourceId: PhysicalResourceId.fromResponse('endpointAddress'),
+          parameters: {
+          "endpointType": "iot:Data-ATS"
+          }
+      },
+      policy: AwsCustomResourcePolicy.fromSdkCalls({resources: AwsCustomResourcePolicy.ANY_RESOURCE})
+    })
+    const IOT_ENDPOINT = ioTEndpoint.getResponseField('endpointAddress')
+    console.log(IOT_ENDPOINT)
```

Comment:
Pls remove, outputs only a token, not resolved during deployment time (`${Token[TOKEN.49]}`)